### DISCUSSION TITLE
Tooling API and Bulk API should not be mutually exclusive for data queries

### DISCUSSION THREAD

**Discussion by FeldhackerChrisPFG** [forcedotcom/cli#2667] [Ideas]
Today (v2.25.7), when executing:
sf data query --use-tooling-api --bulk --query "SELECT RefMetadataComponentName, RefMetadataComponentType, RefMetadataComponentId FROM MetadataComponentDependency WHERE RefMetadataComponentId = '123'"

...results in the error:
"--use-tooling-api=true cannot also be provided when using --bulk"

However, some queries against the Tooling API also support the Bulk API, as with the above query example and as documented here:
https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_metadatacomponentdependency_bulk2_usage.htm

Using the Bulk API with the Tooling API has been supported in the above example since API v49.0.
Support for the Tooling/Bulk combination should be added to the Salesforce CLI as well.

**Comment by AllanOricil** [comment]
Can bulk v1 execute queries using tolling objectd? If not, then the query command can't go back go bulk v1, and because of that it won't be possible to implement pk chunking. Maybe a solution is to make the `--bulk` flag to accept "v1" or "v2", instead of being a boolean. 

https://github.com/forcedotcom/cli/discussions/2463

**Comment by mshanemc** [comment]
update: we don't plan to do to this work prioritize soon.  It turned out to be way more complicated that, "don't make the flags mutually exclusive".  Priority reasoning:

1. dependency API (`MetadataComponentDependency`) is the only thing that uses it so far
2. That's not GA
3. we use jsforce for all the job check/info/etc.  We'd have a lot of changes to make jsforce support tooling API endpoint instead.
